### ISSUE TITLE
Unification of the input interface for new posts, comments and editing comments

### ISSUE THREAD

**Issue by Ravenbird** [friendica/friendica#10485]
Friendica has very different input interfaces for nearly the same tasks, it would make sense to unify this and thus achieve a better user experience. It does not matter if it is the desktop interface or the mobile interface.

Here are a few examples. First, the desktop interface for creating a post:

![](https://s20.directupload.net/images/210710/dgbhkusg.jpg)

Next, the mobile Interface for creating a post:

![](https://s20.directupload.net/images/210710/hj68aexq.jpg)

The interface for creating a comment at the desktop interface:

![](https://s20.directupload.net/images/210710/lvyr5ema.jpg)

The interface for creating a comment at the mobile interface:

![](https://s20.directupload.net/images/210710/vtumqyau.jpg)

If you want to edit a post or comment it looks like this in the desktop interface:

![](https://s20.directupload.net/images/210710/5yfv3cx4.jpg)

If you want to edit a post or comment it looks like this in the mobile interface:

![](https://s20.directupload.net/images/210710/bo4h3adv.jpg)

Why not a single interface for everything. Both in the desktop interface and in the mobile interface. With tabs that you can switch easily and quickly. So about as current in the desktop interface to create posts. Even reasonably current mobile devices should have no problem with this, on the contrary the uniform design would make it easier for everyone.

PS. I hope my 'wonderfull' English is understandable. :-)

**Comment by MrPetovan** [comment]
The main reasons for these display discrepancies is limited width on mobile and the use of a modal window on frio.

Limited screen width on mobile explains:
- The switch from tabs to dropdown list to prevent tabs wrapping on several lines.
- The limited formatting options to prevent formatting buttons wrapping on several lines.

The frio modal explains:
- The image browser moving to a separate tab since it's supposed to load in a modal window as well.
- The preview moving to a separate tab to prevent the modal window to be taller than the screen.

As long as both these reasons will remain, we won't be able to change these behaviors without running into the issues they were addressing.

I've attempted to reconcile both comment and top-level post forms in the Compose page (`/compose` on your node or middle-click on the Post button or the top-right Compose button on the modal window). Let me know what you think about it.

**Comment by annando** [comment]
@MrPetovan the point of @Ravenbird is that he thinks that the composer window of a comment should look like the one of a post - not the differences of the mobile and non-mobile view. And I think that this is a valid point.

**Comment by Ravenbird** [comment]
Hello MrPetovan, you certainly know more about this than I do. I scripted what I needed a bit with Python and that was it. So I see it from the point of view of the normal user, had seen it something like this:

![](https://s20.directupload.net/images/210711/ea7owfev.png)

But of course you can estimate it better than I can. I was just a little concerned with the issue of usability.

**Comment by MrPetovan** [comment]
Speaking of usability, I had to scroll back to your earlier screenshots to remember what was the shield icon for (permissions). So I don't believe icon-only tabs are a good way to convey information on mobile where we can't show tooltips.

**Comment by foss-** [comment]
> Speaking of usability, I had to scroll back to your earlier screenshots to remember what was the shield icon for (permissions). So I don't believe icon-only tabs are a good way to convey information on mobile where we can't show tooltips.

I disagree and would argue that icons are sufficient on mobile. Looking at default mail client on iOS or many other apps it is not uncommon to see icons only.

Maybe a good compromise would be to show the tab name only for active tab.

To me that would be much better than the current dropdown as that hides options and we know that is not good UX. So exposing icons only but adding name to active tab would provide context and allow for better UI exposure and bring mobile closer to desktop UI.

Bonus would be to move "Teilen" or "Send" button to top right while "x" should be "Cancel" and move to the top left. That would follow UI of Mail / Twitter web / Metatext / ... on iOS.

Twitter Web | Metatext
![twitter web](https://user-images.githubusercontent.com/5708172/125646243-cac50a02-ab26-4bf9-9ab7-f8cf0784db90.jpg)![metatext](https://user-images.githubusercontent.com/5708172/125646247-b899e37d-a941-4a0d-86c1-92665fbd73c3.jpg)
